metadata-transformer/src/main/java/no/unit/nva/metadata/extractors/DateExtractor.java

```diff
@@ -0,0 +1,57 @@
+package no.unit.nva.metadata.extractors;
+
+import no.unit.nva.metadata.type.DcTerms;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.PublicationDate;
+import nva.commons.core.JacocoGenerated;
+import org.eclipse.rdf4j.model.Statement;
+
+import java.util.function.Function;
+
+public final class DateExtractor {
+    public static final String DATE_SEPARATOR = "-";
+    public static final int FULL_DATE = 3;
+    public static final int YEAR_ONLY = 1;
+    public static final Function<ExtractionPair, EntityDescription> apply = DateExtractor::extract;
+    public static final int YEAR_PART = 0;
+    public static final int MONTH_PART = 1;
+    public static final int DAY_PART = 2;
+
+
+    @JacocoGenerated
+    private DateExtractor() {
+
+    }
+
+    private static EntityDescription extract(ExtractionPair extractionPair) {
+        if (extractionPair.isDate()) {
+            addDate(extractionPair);
+        }
+        return extractionPair.getEntityDescription();
+    }
+
+    @SuppressWarnings("PMD.CloseResource")
```

Comment:
Is it possible to use try-with-resources to avoid the warning?

Reply:
The places this warning was used previously in this project this was not possible, but this should indeed be checked again.

Reply:
This suppression is not necessary here according to my IDE and PMD

Reply:
We use a different version to Codacy